### ISSUE TITLE
KeyError 'time' when missing workflow variables are added later

### ISSUE THREAD

**Issue by mauritsvanrees** [plone/plone.app.layout#357]
I have Plone 6.0.8 site with a custom workflow where for some reason the standard workflow variables were missing: action, actor, comments, review_history, time.  I decided to add them, but then `@@historyview` gives a traceback:

```
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 181, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 391, in publish_module
  Module ZPublisher.WSGIPublisher, line 285, in publish
  Module ZPublisher.mapply, line 98, in mapply
  Module ZPublisher.WSGIPublisher, line 68, in call_object
  Module plone.app.layout.viewlets.content, line 200, in __call__
  Module Products.Five.browser.pagetemplatefile, line 126, in __call__
  Module Products.Five.browser.pagetemplatefile, line 58, in __call__
  Module zope.pagetemplate.pagetemplate, line 134, in pt_render
  Module Products.PageTemplates.engine, line 365, in __call__
  Module z3c.pt.pagetemplate, line 174, in render
  Module chameleon.zpt.template, line 298, in render
  Module chameleon.template, line 190, in render
  Module 06d4a827c153e365a64dd7fb9d70b6d7, line 183, in render
  Module 1cc249a354d5f31192f69edffbc745d8, line 1027, in render_master
  Module 1cc249a354d5f31192f69edffbc745d8, line 2563, in render_content
  Module 06d4a827c153e365a64dd7fb9d70b6d7, line 152, in __fill_content_core
  Module zope.tales.expressions, line 248, in __call__
  Module Products.PageTemplates.Expressions, line 221, in _eval
  Module Products.PageTemplates.Expressions, line 152, in render
  Module plone.app.layout.viewlets.content, line 529, in __call__
  Module Products.Five.browser.pagetemplatefile, line 126, in __call__
  Module Products.Five.browser.pagetemplatefile, line 58, in __call__
  Module zope.pagetemplate.pagetemplate, line 134, in pt_render
  Module Products.PageTemplates.engine, line 365, in __call__
  Module z3c.pt.pagetemplate, line 174, in render
  Module chameleon.zpt.template, line 298, in render
  Module chameleon.template, line 214, in render
  Module chameleon.utils, line 20, in raise_with_traceback
  Module chameleon.template, line 190, in render
  Module 3d6c204f7e0d9f2ce50033ddfef043d7, line 125, in render
  Module zope.tales.expressions, line 248, in __call__
  Module Products.PageTemplates.Expressions, line 221, in _eval
  Module Products.PageTemplates.Expressions, line 152, in render
  Module plone.app.layout.viewlets.content, line 509, in fullHistory
  Module plone.app.layout.viewlets.content, line 401, in workflowHistory
KeyError: 'action'

 - Expression: "view/fullHistory"
 - Filename:   ... .app.layout/plone/app/layout/viewlets/content_history.pt
 - Location:   (line 2: col 15)
 - Source:     history view/fullHistory;
                       ^^^^^^^^^^^^^^^^
```

I think it would be good to be forgiving here when accessing the variables.
I am making a PR.

**Comment by mauritsvanrees** [comment]
Ah frac, I committed directly to master...  Sorry!

I think it is fine, and the tests passed locally.  But anyone should feel free to revert, and I will open a PR instead.

**Comment by davisagli** [comment]
@mauritsvanrees LGTM

**Comment by mauritsvanrees** [comment]
> @mauritsvanrees LGTM

@davisagli Thanks for checking!

**Comment by mauritsvanrees** [comment]
Sigh, since I directly committed to master instead of via a merged PR, `plone.app.layout` was not added to `checkouts.cfg`, so I never released this and it is not in Plone 6.0.9...

I now released 4.0.8 with this fix.
